specs/DesignDiscussions/DotNetTemplate.md

```diff
@@ -0,0 +1,338 @@
+# Overview
+## Templating
+![](img1.png)
+
+Template allows users to define where data binding occurs.
+
+```json
+{
+    "type": "TextBlock",
+    "text": "{latestPrice} ",
+    "size": "ExtraLarge"
+}
+```
+
+Once binding is defined, adaptive cards will be updated by changing only the data.
+```json
+{
+  "latestPrice": 128.9,
+}
+```
+### Syntax
+- Binding syntax starts with { and ends with }. E.g., {myProperty}
+- Dot-notation to access sub-objects
+ - {employee.manager.name}
+- Indexer syntax to retrieve properties by key or items in an array
+ - {employee.peers[0].name}
+- $data assigns data context to current element
+- $when conditional expression that determines if current element to be included.
+- functions are provided to help SDK users
+
+# Parsing
+We use ANTLR to generate parser and lexer
+
+## ANTLR
+From a grammar, ANTLR generates a parser that can build parse trees and also generates a listener interface (or visitor) that makes it easy to respond to the recognition of phrases of interest.
+ANTLR is really two things: a tool that translates your grammar to a parser/lexer in Java (or other target language) and the runtime needed by the generated parsers/lexers.
```

Comment:
>or other target language [](start = 91, length = 24)

Question in the room: What languages do we need? Do we need objective C? This will be happening in a C++ layer and we won't need objective C.

Current plan is for .NET and JavaScript to handle server scenarios. Long term we'll want to support client side for mobile/UWP as well. It won't be a problem to handle this in C++ and consume it on all shared model platforms.